File: src/types/hardware.ts
// This file mirrors Rust models/hardware.rs types for the frontend.

export interface SystemInfo {
  os_name: string;
  os_version: string;
  hostname: string;
  cpu: CpuInfo;
  gpus: GpuInfo[];
  total_memory_bytes: number;
  disks: DiskInfo[];
}

export interface CpuInfo {
  brand: string;
  vendor: string;
  core_count: number;
  thread_count: number;
  base_frequency_mhz: number;
}

export interface GpuInfo {
  name: string;
  vendor: string;
  vram_total_mb: number;
  driver_version: string;
  is_nvidia: boolean;
}

export interface DiskInfo {
  name: string;
  mount_point: string;
  filesystem: string;
  total_bytes: number;
  is_removable: boolean;
  disk_type: string;
}

export interface SensorSnapshot {
  timestamp: number;
  cpu: CpuSensorData;
  gpus: GpuSensorData[];
  memory: MemorySensorData;
  disks: DiskSensorData[];
}

export interface CpuSensorData {
  usage_percent: number;
  temperature_c: number | null;
  frequency_mhz: number;
  per_core_usage: number[];
  power_watts: number | null;
}

export interface GpuSensorData {
  name: string;
  usage_percent: number;
  temperature_c: number | null;
  vram_used_mb: number;
  vram_total_mb: number;
  fan_speed_percent: number | null;
  power_watts: number | null;
  clock_core_mhz: number | null;
  clock_memory_mhz: number | null;
}

export interface MemorySensorData {
  used_bytes: number;
  total_bytes: number;
  usage_percent: number;
  available_bytes: number;
  swap_used_bytes: number;
  swap_total_bytes: number;
}

export interface DiskSensorData {
  name: string;
  used_bytes: number;
  total_bytes: number;
  usage_percent: number;
  read_bytes_per_sec: number;
  write_bytes_per_sec: number;
  temperature_c: number | null;
}

File: src-tauri/src/models/hardware.rs
use serde::{Deserialize, Serialize};

/// 系统静态信息（一次性获取）
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SystemInfo {
    pub os_name: String,
    pub os_version: String,
    pub hostname: String,
    pub cpu: CpuInfo,
    pub gpus: Vec<GpuInfo>,
    pub total_memory_bytes: u64,
    pub disks: Vec<DiskInfo>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CpuInfo {
    pub brand: String,
    pub vendor: String,
    pub core_count: usize,
    pub thread_count: usize,
    pub base_frequency_mhz: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct GpuInfo {
    pub name: String,
    pub vendor: String,
    pub vram_total_mb: u64,
    pub driver_version: String,
    pub is_nvidia: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct DiskInfo {
    pub name: String,
    pub mount_point: String,
    pub filesystem: String,
    pub total_bytes: u64,
    pub is_removable: bool,
    pub disk_type: String,
}

/// 实时传感器快照（周期性推送）
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SensorSnapshot {
    pub timestamp: u64,
    pub cpu: CpuSensorData,
    pub gpus: Vec<GpuSensorData>,
    pub memory: MemorySensorData,
    pub disks: Vec<DiskSensorData>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CpuSensorData {
    pub usage_percent: f32,
    pub temperature_c: Option<f32>,
    pub frequency_mhz: f64,
    pub per_core_usage: Vec<f32>,
    pub power_watts: Option<f32>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct GpuSensorData {
    pub name: String,
    pub usage_percent: f32,
    pub temperature_c: Option<f32>,
    pub vram_used_mb: u64,
    pub vram_total_mb: u64,
    pub fan_speed_percent: Option<f32>,
    pub power_watts: Option<f32>,
    pub clock_core_mhz: Option<u32>,
    pub clock_memory_mhz: Option<u32>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MemorySensorData {
    pub used_bytes: u64,
    pub total_bytes: u64,
    pub usage_percent: f32,
    pub available_bytes: u64,
    pub swap_used_bytes: u64,
    pub swap_total_bytes: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct DiskSensorData {
    pub name: String,
    pub used_bytes: u64,
    pub total_bytes: u64,
    pub usage_percent: f32,
    pub read_bytes_per_sec: u64,
    pub write_bytes_per_sec: u64,
    pub temperature_c: Option<f32>,
}
